Modules/Scripted/DICOMLib/DICOMUtils.py

```diff
@@ -906,6 +912,7 @@ def onNodeAdded(caller, event, calldata):
             break
 
     slicer.mrmlScene.RemoveObserver(sceneObserverTag)
+    slicer.mrmlScene.RemoveObserver(sceneObserverTag2)
```

Comment:
It would be nice to if we could ensure that the observers are removed even if an error occurs. For example, by adding a try/finally block.